documentation/config_file_auth.md

```diff
@@ -9,142 +9,168 @@ canonical: "/puppetserver/latest/config_file_auth.html"
-type: path
-~~~
+### `allow-header-cert-info`
 
-> **Note:** Enclose the path value within double quotes. While the HOCON format does not always require wrapping string values with double quotes, special characters commonly used in regular expressions (such as `*`) break HOCON parsing unless the entire value is surrounded by double quotes.
+> **Note:** Puppet Server ignores the setting of the same name in [`master.conf`](./config_file_master.html) in favor of this setting in the new `auth.conf` file. If you use the deprecated authentication method and legacy [Puppet `auth.conf`][] file, you must instead configure this setting in `master.conf`.
 
-In the deprecated `auth.conf` format, start the `path` value with a tilde (`~`) character to indicate that it contains a regular expression.
+This optional `authorization` section parameter determines whether to enable [external SSL termination](./external_ssl_termination.html) on all HTTP endpoints that Puppet Server handles, including those served by the "master" service, the certificate authority API, and the Puppet Admin API. It also controls how Puppet Server derives the user's identity for authorization purposes. The default value is `false`.
 
-~~~
-# Regular expression to match a path in a URL.
-path ~ ^/puppet/v3/report/([^/]+)$
+If this setting is `true`, Puppet Server ignores any presented certificate and relies completely on header data to authorize requests. 
 
-# Literal string to match at the start of a URL's path.
-path /puppet/v3/report/
-~~~
+> **Warning!** This is very insecure; **do not enable this parameter** unless you've secured your network to prevent **any** untrusted access to Puppet Server.
 
-### `method`
+You cannot rename any of the `X-Client` headers when this setting is enabled, and you must specified identity through the `X-Client-Verify`, `X-Client-DN`, and `X-Client-Cert` headers.
 
-If a rule contains the `method` parameter, it only applies to requests that use the value's corresponding HTTP methods.
+For more information, see [External SSL Termination](./external_ssl_termination.html#disable-https-for-puppet-server) in the Puppet Server documentation and [Configuring the Authorization Service](https://github.com/puppetlabs/trapperkeeper-authorization/blob/master/doc/authorization-config.md#allow-header-cert-info) in the `trapperkeeper-authorization` documentation.
 
-In a HOCON `auth.conf` rule, list the HTTP methods.
+### `rules`
 
-~~~
-# Use GET and POST.
-method: [get, post]
+The required `rules` array of a Puppet Server's HOCON `auth.conf` file determines how Puppet Server responds to a request. Each element is a map of settings pertaining to a rule, and when Puppet Server receives a request, it evaluates that request against each rule looking for a match.
 
-# Use PUT.
-method: put
-~~~
+You define each rule by adding parameters to the rule's [`match-request`](#match-request) section. A `rules` array can contain as many `match-request` sections as you need.
```

Comment:
No, `match-request` is required (https://github.com/puppetlabs/trapperkeeper-authorization/blob/master/doc/authorization-config.md#match-request) (https://github.com/puppetlabs/trapperkeeper-authorization/blob/master/src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj#L139-L141)